test/unit_tests/controllers/auth_controller_test.dart

```diff
@@ -6,13 +6,27 @@ import 'package:pispapp/repositories/firebase/auth_repository.dart';
 
 class MockAuthRepository extends Mock implements AuthRepository {}
 
+/// NOTE: We are overriding only the part of the controller that is dependent
+/// on Firestore. We are still testing the other implementations in
+/// [AuthController].
+class TestAuthController extends AuthController {
+  TestAuthController(AuthRepository authRepository) : super(authRepository);
+
+  @override
+  Future<void> createUserDataControllerAndCreateUserEntity(User u) async {
+    // do nothing in tests
```

Comment:
if we're doing nothing then why have the test here?

Reply:
this isn't a test itself, it's part of the mock. I am mocking the controller being tested but only overriding 1 of the functions which is tightly coupled with firebase. 

The rest of the controller is being tested as normal